### DISCUSSION TITLE
db:seed fails, tests wont run for development version linked to latest dev commit 8407c23c

### DISCUSSION THREAD

**Discussion by BenjaminOsei** [cypress-io/cypress-realworld-app#1132] [Q&A]
Goodevening,

I'm having the same db connection issue as described under discussion #1032.
I too got below error after running 'npm run start:api' within my terminal:

------------------------------------------------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------------------------------------------------

> cypress-realworld-app@1.0.0 start:api C:\Users\benja\source\repos\TheCypressRealWorldApp\cypress-realworld-app
> yarn tsnode --files backend/app.ts

yarn run v1.22.17
$ nyc --silent ts-node -P tsconfig.tsnode.json --files backend/app.ts
C:\Users\benja\source\repos\TheCypressRealWorldApp\cypress-realworld-app\node_modules\ts-node\src\index.ts:692
    return new TSError(diagnosticText, diagnosticCodes);
           ^
TSError: ⨯ Unable to compile TypeScript:
backend/app.ts:79:11 - error TS2769: No overload matches this call.
  The last overload gave the following error.
    Argument of type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to parameter of type 'PathParams'.
      Type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is missing the following properties from type '(string | RegExp)[]': pop, push, concat, join, and 27 more.

79   app.use(checkAuth0Jwt);
             ~~~~~~~~~~~~~

  node_modules/@types/express-serve-static-core/index.d.ts:61:5
    61     <P extends Params = ParamsDictionary, ResBody = any, ReqBody = any>(path: PathParams, ...handlers: Array<RequestHandlerParams<P, ResBody, ReqBody>>): T;
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    The last overload is declared here.
backend/app.ts:89:11 - error TS2769: No overload matches this call.
  The last overload gave the following error.
    Argument of type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to parameter of type 'PathParams'.
      Type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to 
type '(string | RegExp)[]'.

89   app.use(checkCognitoJwt);
             ~~~~~~~~~~~~~~~

  node_modules/@types/express-serve-static-core/index.d.ts:61:5
    61     <P extends Params = ParamsDictionary, ResBody = any, ReqBody = any>(path: PathParams, ...handlers: Array<RequestHandlerParams<P, ResBody, ReqBody>>): T;
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    The last overload is declared here.
backend/app.ts:94:11 - error TS2769: No overload matches this call.
  The last overload gave the following error.
    Argument of type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to parameter of type 'PathParams'.
      Type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to 
type '(string | RegExp)[]'.

94   app.use(checkGoogleJwt);
             ~~~~~~~~~~~~~~

  node_modules/@types/express-serve-static-core/index.d.ts:61:5
    61     <P extends Params = ParamsDictionary, ResBody = any, ReqBody = any>(path: PathParams, ...handlers: Array<RequestHandlerParams<P, ResBody, ReqBody>>): T;
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    The last overload is declared here.

------------------------------------------------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------------------------------------------------

And this also resulted in below result with the Cypress Test Runner:

![image](https://user-images.githubusercontent.com/45737274/139561669-2006a5b0-2897-4484-936b-86ec019e1236.png)

Could someone help me out with this please?

Gr. Benjamin

**Comment by kevinold** [comment]
@BenjaminOsei Do you have the issue if you run `yarn dev`?

**Comment by BenjaminOsei** [comment]
Hi Kevin, 
I don't know why, but yarn is not recognised by my system. 
When I enter 'yarn dev', I get the following error message:
![image](https://user-images.githubusercontent.com/45737274/139857915-c8b660be-873f-4976-9c18-4e9e8c86a5f3.png)
I already tried fixing this error by entering the following:
![image](https://user-images.githubusercontent.com/45737274/139858555-6414036d-6693-4e56-996d-ac11e515b113.png)

Set-ExecutionPolicy -ExecutionPolicy Unrestricted -Scope LocalMachine

![image](https://user-images.githubusercontent.com/45737274/139859626-16f28f7e-979c-4bd9-b065-2d94fa8207a3.png)

yarn dev now does work and both the api as the react front end got started now. I'll have to update on running the test successfully on a later moment though. Please bear with me Kevin, I'll get back to this...
for now this was my result:
![image](https://user-images.githubusercontent.com/45737274/139861210-42ad8a36-acd0-472e-9c0c-9d384f4f7e48.png)

**Comment by BenjaminOsei** [comment]
Hi Kevin, even after yarn dev I couldn't run a test as expected. after having submitted yarn dev, I got the following logs:

PS C:\Users\benja\source\repos\TheCypressRealWorldApp\cypress-realworld-app> yarn dev
yarn run v1.22.17
$ yarn db:seed:dev
$ yarn predev:cognito:ci
$ yarn copy:mock:awsexports && yarn copy:mock:awsexportses5
$ ncp scripts/mock-aws-exports.js src/aws-exports.js
$ ncp scripts/mock-aws-exports-es5.js aws-exports-es5.js
$ ncp ./data/database-seed.json ./data/database.json
$ cross-env NODE_ENV=development concurrently yarn:start:react yarn:start:api:watch
$ react-scripts -r @cypress/instrument-cra start
$ nodemon --exec yarn tsnode --watch 'backend' backend/app.ts
[start:api:watch] [nodemon] 2.0.13
[start:api:watch] [nodemon] to restart at any time, enter `rs`
[start:api:watch] [nodemon] watching path(s): 'backend'
[start:api:watch] [nodemon] watching extensions: ts,json
[start:api:watch] [nodemon] starting `yarn tsnode backend/app.ts`
$ nyc --silent ts-node -P tsconfig.tsnode.json backend/app.ts
[start:api:watch] C:\Users\benja\source\repos\TheCypressRealWorldApp\cypress-realworld-app\node_modules\ts-node\src\index.ts:692
[start:api:watch]     return new TSError(diagnosticText, diagnosticCodes);
[start:api:watch]            ^
[start:api:watch] TSError: ⨯ Unable to compile TypeScript:
[start:api:watch] backend/app.ts(79,11): error TS2769: No overload matches this call.
[start:api:watch]   The last overload gave the following error.
[start:api:watch]     Argument of type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to parameter of type 'PathParams'.
[start:api:watch]       Type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is missing the following properties from type '(string | RegExp)[]': pop, push, concat, join, and 27 more.
[start:api:watch] backend/app.ts(89,11): error TS2769: No overload matches this call.
[start:api:watch]   The last overload gave the following error.
[start:api:watch]     Argument of type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to parameter of type 'PathParams'.
[start:api:watch]       Type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to type '(string | RegExp)[]'.
[start:api:watch] backend/app.ts(94,11): error TS2769: No overload matches this call.
[start:api:watch]   The last overload gave the following error.
[start:api:watch]     Argument of type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to parameter of type 'PathParams'.
[start:api:watch]       Type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to type '(string | RegExp)[]'.
[start:api:watch]
[start:api:watch]     at createTSError (C:\Users\benja\source\repos\TheCypressRealWorldApp\cypress-realworld-app\node_modules\ts-node\src\index.ts:692:12)
[start:api:watch]     at reportTSError (C:\Users\benja\source\repos\TheCypressRealWorldApp\cypress-realworld-app\node_modules\ts-node\src\index.ts:696:19)
[start:api:watch]     at getOutput (C:\Users\benja\source\repos\TheCypressRealWorldApp\cypress-realworld-app\node_modules\ts-node\src\index.ts:883:36)
[start:api:watch]     at Object.compile (C:\Users\benja\source\repos\TheCypressRealWorldApp\cypress-realworld-app\node_modules\ts-node\src\index.ts:1185:30)
[start:api:watch]     at Module.m._compile (C:\Users\benja\source\repos\TheCypressRealWorldApp\cypress-realworld-app\node_modules\ts-node\src\index.ts:1309:30)
[start:api:watch]     at module.exports (C:\Users\benja\source\repos\TheCypressRealWorldApp\cypress-realworld-app\node_modules\default-require-extensions\js.js:7:9)
[start:api:watch]     at C:\Users\benja\source\repos\TheCypressRealWorldApp\cypress-realworld-app\node_modules\append-transform\index.js:64:4
[start:api:watch]     at require.extensions.<computed> (C:\Users\benja\source\repos\TheCypressRealWorldApp\cypress-realworld-app\node_modules\ts-node\src\index.ts:1313:12)
[start:api:watch]     at Object.<anonymous> (C:\Users\benja\source\repos\TheCypressRealWorldApp\cypress-realworld-app\node_modules\append-transform\index.js:64:4)
[start:api:watch]     at Module.load (internal/modules/cjs/loader.js:950:32) {
[start:api:watch]   diagnosticText: 'backend/app.ts(79,11): error TS2769: No overload matches this call.\r\n' +
[start:api:watch]     '  The last overload gave the following error.\r\n' +
[start:api:watch]     "    Argument of type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to parameter of type 'PathParams'.\r\n" +
[start:api:watch]     "      Type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is missing the following properties from type '(string | RegExp)[]': pop, push, concat, join, and 27 more.\r\n" +     
[start:api:watch]     'backend/app.ts(89,11): error TS2769: No overload matches this call.\r\n' +
[start:api:watch]     '  The last overload gave the following error.\r\n' +
[start:api:watch]     "    Argument of type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to parameter of type 'PathParams'.\r\n" +
[start:api:watch]     "      Type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to type '(string | RegExp)[]'.\r\n" +
[start:api:watch]     'backend/app.ts(94,11): error TS2769: No overload matches this call.\r\n' +
[start:api:watch]     '  The last overload gave the following error.\r\n' +
[start:api:watch]     "    Argument of type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to parameter of type 'PathParams'.\r\n" +
[start:api:watch]     "      Type 'RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to type '(string | RegExp)[]'.\r\n",
[start:api:watch]   diagnosticCodes: [ 2769, 2769, 2769 ]
[start:api:watch] }
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
[start:api:watch] [nodemon] app crashed - waiting for file changes before starting...
[start:react] [HPM] Proxy created: [ '/login', '/callback', '/logout', '/checkAuth', 'graphql' ]  ->  http://localhost:undefined
[start:react] [HPM] Subscribed to http-proxy events:  [ 'error', 'close' ]
[start:react] i ｢wds｣: Project is running at http://192.168.204.1/
[start:react] i ｢wds｣: webpack output is served from
[start:react] i ｢wds｣: Content not from webpack is served from C:\Users\benja\source\repos\TheCypressRealWorldApp\cypress-realworld-app\public
[start:react] i ｢wds｣: 404s will fallback to /
[start:react] Starting the development server...
[start:react]
[start:react] Compiled successfully!
[start:react]
[start:react] You can now view cypress-realworld-app in the browser.
[start:react]
[start:react]   Local:            http://localhost:3000
[start:react]   On Your Network:  http://192.168.204.1:3000
[start:react]
[start:react] Note that the development build is not optimized.
[start:react] To create a production build, use yarn build.
[start:react]
[start:react] Compiling...
[start:react] Compiled successfully!

--------------------------------------------------------------------------------------------------------------------------------------------

Though I nevertheless tried running the test from the testrunner via yarn cypress:open, I indeed still got the following page:
![image](https://user-images.githubusercontent.com/45737274/139963629-37db5f85-b74a-458a-8c0b-abb05ca436f2.png)

Do you have any ideas on what I still could try? 

With kind regards,
Benjamin

**Reply by BenjaminOsei** [reply]
Hi @kevinold, have you had a chance to look at this issue? I'd really appreciate any help on this issue

Gr. Benjamin
